cdap-kubernetes/src/e2e-test/java/io/cdap/cdap/master/environment/k8s/stepsdesign/KubernetesNamespaceCreation.java

```
@@ -0,0 +1,120 @@
+  }
+
+  @Then("Go to next page in namespace creation wizard")
+  public static void nextNamespaceCreationPage() {
+    NamespaceCreationActions.clickNextButton();
+  }
+
+  @Then("Finish namespace creation")
+  public static void finishNamespaceCreation() {
+    NamespaceCreationActions.clickFinishButton();
+  }
+
+  @Then("Verify Kubernetes namespace {string} exists")
+  public static void verifyKubeNamespaceExists(String namespace) throws IOException {
+    try {
+      coreV1Api.readNamespace(namespace, null, null, null);
+    } catch (ApiException e) {
+      throw new IOException("Error occurred while checking for Kubernetes namespace. Error code = "
+                              + e.getCode() + ", Body = " + e.getResponseBody(), e);
+    }
+  }
+
+  @Then("Verify CPU limit {string} and memory limit {string} exists for {string}")
+  public static void verifyResourceLimits(String cpuLimit, String memLimit, String namespace) throws IOException {
+    Map<String, Quantity> hardLimitMap = ImmutableMap.of("limits.cpu", new Quantity(cpuLimit), "limits.memory",
+                                                         new Quantity(memLimit));
+    try {
+      V1ResourceQuota resourceQuota = coreV1Api.readNamespacedResourceQuota(KubeMasterEnvironment.RESOURCE_QUOTA_NAME,
+                                                                            namespace, null, null, null);
+      if (!hardLimitMap.equals(resourceQuota.getSpec().getHard())) {
+        throw new IOException(String.format("Incorrect resource limits. Expected %s but received %s",
+                                            hardLimitMap, resourceQuota.getSpec().getHard()));
+      }
+    } catch (ApiException e) {
+      throw new IOException("Error occurred while checking for Kubernetes resource quota. Error code = "
+                              + e.getCode() + ", Body = " + e.getResponseBody(), e);
+    } catch (NullPointerException e) {
```

Where is NullPointerException thrown? Can we just add a null check and throw IOException when we there's an unexpected null value instead of catching NullpointerException?

---

fixed